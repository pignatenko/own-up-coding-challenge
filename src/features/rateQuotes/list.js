import { DataGrid } from "@material-ui/data-grid";
import Alert from "@material-ui/lab/Alert";
import { useSelector } from "react-redux";

function* numberFormat(original) {
  let number = original.toString();

  while (number && number.length) {
    yield number.slice(-3);
    number = number.slice(0, -3);
  }
}

function displayDollarAmount(params) {
  if (!params.value) {
    return;
  }

  const [dollars] = params.value.toFixed(2).toString().split(".", 2);
  const generator = numberFormat(dollars);
  const breakout = Array.from(generator);

  return `$${breakout.reverse().join(",")}`;
}

function displayPercentage(params) {
  if (!params.value) {
    return;
  }
  return `%${params.value}`;
}

function displayErrors(errors) {
  return errors.map((error, idx) => (
    <Alert key={idx} severity="error">
      {error}
    </Alert>
  ));
}

export function RateQuoteList() {
  const { rateQuoteErrors, rateQuotes } = useSelector((s) => s.rateQuotes);

  const columns = [
    { field: "lenderName", headerName: "Lender Name", width: 250 },
    { field: "loanType", headerName: "Product", width: 250 },
    {
      field: "interestRate",
      headerName: "Rate",
      width: 100,
      valueFormatter: displayPercentage,
    },
    {
      field: "closingCosts",
      headerName: "Closing Costs",
      width: 200,
      valueFormatter: displayDollarAmount,
    },
    {
      field: "monthlyPayment",
      headerName: "Monthly Payment",
      width: 200,
      valueFormatter: displayDollarAmount,
    },
    {
      field: "apr",
      headerName: "APR",
      width: 100,
      valueFormatter: displayPercentage,
    },
  ];

  return (
    <div className="RateQuoteList">
      {displayErrors(rateQuoteErrors)}

      <DataGrid autoHeight hideFooter columns={columns} rows={rateQuotes} />
    </div>
  );
}
