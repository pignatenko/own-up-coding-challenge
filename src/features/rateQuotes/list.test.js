import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { createMount } from  '@material-ui/core/test-utils';
import { RateQuoteList } from './list';
import {renderToJson} from 'enzyme-to-json';
import rateQuotesFixture from '../../fixtures/rateQuotesFixture.json';

const MockStore = configureStore([]);

describe("<RateQuoteList />", () => {
  let initialState;
  let component;
  let store
  let mount;

  beforeAll(()=> {
    mount = createMount();
  });

  describe("when there are errors", () => {
    beforeEach(() => {
      initialState = {
        rateQuoteErrors: ['first error', 'second error', 'third error'],
        rateQuotes: [],
      };
      store = MockStore({rateQuotes: initialState});
      component = mount(<Provider store={store}> <RateQuoteList/> </Provider>);
    });

    it('matches the snapshot', () => {
      expect(renderToJson(component.render())).toMatchSnapshot(); 
    });
  });

  describe('when there are rate quotes to show', () => {
    beforeEach(() => {
      initialState = {
        rateQuoteErrors: [],
        rateQuotes: rateQuotesFixture
      };
      store = MockStore({rateQuotes: initialState});
      component = mount(<Provider store={store}> <RateQuoteList/> </Provider>);
    });

    it('matches the snapshot', () => {
      expect(renderToJson(component.render())).toMatchSnapshot(); 
    });
  });
});
