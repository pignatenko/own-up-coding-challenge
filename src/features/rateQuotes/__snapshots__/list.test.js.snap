// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RateQuoteList /> when there are errors matches the snapshot 1`] = `
Array [
  " ",
  <div
    class="RateQuoteList"
  >
    <div
      class="MuiPaper-root MuiAlert-root MuiAlert-standardError MuiPaper-elevation0"
      role="alert"
    >
      <div
        class="MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message"
      >
        first error
      </div>
    </div>
    <div
      class="MuiPaper-root MuiAlert-root MuiAlert-standardError MuiPaper-elevation0"
      role="alert"
    >
      <div
        class="MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message"
      >
        second error
      </div>
    </div>
    <div
      class="MuiPaper-root MuiAlert-root MuiAlert-standardError MuiPaper-elevation0"
      role="alert"
    >
      <div
        class="MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message"
      >
        third error
      </div>
    </div>
    <div
      style="overflow: visible; height: 0px; width: 0px;"
    >
      <div
        aria-colcount="6"
        aria-label="grid"
        aria-multiselectable="false"
        aria-rowcount="0"
        class="MuiDataGrid-root MuiDataGrid-root"
        role="grid"
        style="width: 0px; height: 160px;"
        tabindex="0"
      >
        <div />
        <div
          class="MuiDataGrid-mainGridContainer"
        >
          <div
            class="MuiDataGrid-columnsContainer"
            style="min-height: 56px; max-height: 56px; line-height: 56px;"
          >
            <div
              aria-rowindex="1"
              class="MuiDataGrid-colCellWrapper scroll"
              role="row"
              style="transform: translate3d(-0px, 0, 0); min-width: 1100px;"
            >
              <div
                aria-colindex="1"
                class="MuiDataGrid-colCell MuiDataGrid-colCellSortable"
                data-field="lenderName"
                role="columnheader"
                style="width: 250px; min-width: 250px; max-width: 250px;"
                tabindex="-1"
              >
                <div
                  class="MuiDataGrid-colCell-draggable"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-colCellTitleContainer"
                  >
                    <div
                      aria-label="Lender Name"
                      class="MuiDataGrid-colCellTitle"
                      title=""
                    >
                      Lender Name
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-menuIcon"
                  >
                    <button
                      aria-label="Menu"
                      class="MuiButtonBase-root MuiIconButton-root MuiDataGrid-menuIconButton MuiIconButton-sizeSmall"
                      tabindex="0"
                      title="Menu"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator"
                  style="min-height: 56px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiDataGrid-iconSeparator"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
              <div
                aria-colindex="2"
                class="MuiDataGrid-colCell MuiDataGrid-colCellSortable"
                data-field="loanType"
                role="columnheader"
                style="width: 250px; min-width: 250px; max-width: 250px;"
                tabindex="-1"
              >
                <div
                  class="MuiDataGrid-colCell-draggable"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-colCellTitleContainer"
                  >
                    <div
                      aria-label="Product"
                      class="MuiDataGrid-colCellTitle"
                      title=""
                    >
                      Product
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-menuIcon"
                  >
                    <button
                      aria-label="Menu"
                      class="MuiButtonBase-root MuiIconButton-root MuiDataGrid-menuIconButton MuiIconButton-sizeSmall"
                      tabindex="0"
                      title="Menu"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator"
                  style="min-height: 56px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiDataGrid-iconSeparator"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
              <div
                aria-colindex="3"
                class="MuiDataGrid-colCell MuiDataGrid-colCellSortable"
                data-field="interestRate"
                role="columnheader"
                style="width: 100px; min-width: 100px; max-width: 100px;"
                tabindex="-1"
              >
                <div
                  class="MuiDataGrid-colCell-draggable"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-colCellTitleContainer"
                  >
                    <div
                      aria-label="Rate"
                      class="MuiDataGrid-colCellTitle"
                      title=""
                    >
                      Rate
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-menuIcon"
                  >
                    <button
                      aria-label="Menu"
                      class="MuiButtonBase-root MuiIconButton-root MuiDataGrid-menuIconButton MuiIconButton-sizeSmall"
                      tabindex="0"
                      title="Menu"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator"
                  style="min-height: 56px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiDataGrid-iconSeparator"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="MuiDataGrid-cell MuiDataGrid-cellLeft"
                role="cell"
                style="min-width: 500px; max-width: 500px; line-height: 55px; min-height: 56px; max-height: 56px;"
              />
            </div>
          </div>
          <div
            class="MuiDataGrid-overlay"
            style="top: 56px;"
          >
            No Rows
          </div>
          <div
            class="MuiDataGrid-window"
            style="top: 56px; overflow-y: hidden;"
          >
            <div
              class="MuiDataGrid-dataContainer data-container"
              style="min-height: 15px; min-width: 1100px;"
            >
              <div
                class="MuiDataGrid-viewport"
                style="min-width: 0; max-width: 0; min-height: 0; max-height: 0;"
              >
                <div
                  class="rendering-zone"
                  style="max-height: 67px; width: 1100px; pointer-events: none; transform: translate3d(-0px, -0px, 0);"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Mui-resizeTriggers"
    >
      <div
        class="expand-trigger"
      >
        <div
          style="width: 1px; height: 1px;"
        />
      </div>
      <div
        class="contract-trigger"
      />
    </div>
  </div>,
  " ",
]
`;

exports[`<RateQuoteList /> when there are rate quotes to show matches the snapshot 1`] = `
Array [
  " ",
  <div
    class="RateQuoteList"
  >
    <div
      style="overflow: visible; height: 0px; width: 0px;"
    >
      <div
        aria-colcount="6"
        aria-label="grid"
        aria-multiselectable="false"
        aria-rowcount="3"
        class="MuiDataGrid-root MuiDataGrid-root"
        role="grid"
        style="width: 0px; height: 227px;"
        tabindex="0"
      >
        <div />
        <div
          class="MuiDataGrid-mainGridContainer"
        >
          <div
            class="MuiDataGrid-columnsContainer"
            style="min-height: 56px; max-height: 56px; line-height: 56px;"
          >
            <div
              aria-rowindex="1"
              class="MuiDataGrid-colCellWrapper scroll"
              role="row"
              style="transform: translate3d(-0px, 0, 0); min-width: 1100px;"
            >
              <div
                aria-colindex="1"
                class="MuiDataGrid-colCell MuiDataGrid-colCellSortable"
                data-field="lenderName"
                role="columnheader"
                style="width: 250px; min-width: 250px; max-width: 250px;"
                tabindex="-1"
              >
                <div
                  class="MuiDataGrid-colCell-draggable"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-colCellTitleContainer"
                  >
                    <div
                      aria-label="Lender Name"
                      class="MuiDataGrid-colCellTitle"
                      title=""
                    >
                      Lender Name
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-menuIcon"
                  >
                    <button
                      aria-label="Menu"
                      class="MuiButtonBase-root MuiIconButton-root MuiDataGrid-menuIconButton MuiIconButton-sizeSmall"
                      tabindex="0"
                      title="Menu"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator"
                  style="min-height: 56px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiDataGrid-iconSeparator"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
              <div
                aria-colindex="2"
                class="MuiDataGrid-colCell MuiDataGrid-colCellSortable"
                data-field="loanType"
                role="columnheader"
                style="width: 250px; min-width: 250px; max-width: 250px;"
                tabindex="-1"
              >
                <div
                  class="MuiDataGrid-colCell-draggable"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-colCellTitleContainer"
                  >
                    <div
                      aria-label="Product"
                      class="MuiDataGrid-colCellTitle"
                      title=""
                    >
                      Product
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-menuIcon"
                  >
                    <button
                      aria-label="Menu"
                      class="MuiButtonBase-root MuiIconButton-root MuiDataGrid-menuIconButton MuiIconButton-sizeSmall"
                      tabindex="0"
                      title="Menu"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator"
                  style="min-height: 56px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiDataGrid-iconSeparator"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
              <div
                aria-colindex="3"
                class="MuiDataGrid-colCell MuiDataGrid-colCellSortable"
                data-field="interestRate"
                role="columnheader"
                style="width: 100px; min-width: 100px; max-width: 100px;"
                tabindex="-1"
              >
                <div
                  class="MuiDataGrid-colCell-draggable"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-colCellTitleContainer"
                  >
                    <div
                      aria-label="Rate"
                      class="MuiDataGrid-colCellTitle"
                      title=""
                    >
                      Rate
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-menuIcon"
                  >
                    <button
                      aria-label="Menu"
                      class="MuiButtonBase-root MuiIconButton-root MuiDataGrid-menuIconButton MuiIconButton-sizeSmall"
                      tabindex="0"
                      title="Menu"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator"
                  style="min-height: 56px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiDataGrid-iconSeparator"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="MuiDataGrid-cell MuiDataGrid-cellLeft"
                role="cell"
                style="min-width: 500px; max-width: 500px; line-height: 55px; min-height: 56px; max-height: 56px;"
              />
            </div>
          </div>
          <div
            class="MuiDataGrid-window"
            style="top: 56px; overflow-y: hidden;"
          >
            <div
              class="MuiDataGrid-dataContainer data-container"
              style="min-height: 171px; min-width: 1100px;"
            >
              <div
                class="MuiDataGrid-viewport"
                style="min-width: 0; max-width: 0; min-height: 156px; max-height: 156px;"
              >
                <div
                  class="rendering-zone"
                  style="max-height: 379px; width: 1100px; pointer-events: none; transform: translate3d(-0px, -0px, 0);"
                >
                  <div
                    aria-rowindex="2"
                    aria-selected="false"
                    class="MuiDataGrid-row Mui-even"
                    data-id="1"
                    data-rowindex="0"
                    role="row"
                    style="max-height: 52px; min-height: 52px;"
                  >
                    <div
                      aria-colindex="0"
                      class="MuiDataGrid-cell MuiDataGrid-cellLeft"
                      data-field="lenderName"
                      data-rowindex="0"
                      data-value="TFB Federal Credit Union"
                      role="cell"
                      style="min-width: 250px; max-width: 250px; line-height: 51px; min-height: 52px; max-height: 52px;"
                      tabindex="0"
                    >
                      TFB Federal Credit Union
                    </div>
                    <div
                      aria-colindex="1"
                      class="MuiDataGrid-cell MuiDataGrid-cellLeft"
                      data-field="loanType"
                      data-rowindex="0"
                      data-value="7/1 ARM"
                      role="cell"
                      style="min-width: 250px; max-width: 250px; line-height: 51px; min-height: 52px; max-height: 52px;"
                      tabindex="-1"
                    >
                      7/1 ARM
                    </div>
                    <div
                      aria-colindex="2"
                      class="MuiDataGrid-cell MuiDataGrid-cellLeft"
                      data-field="interestRate"
                      data-rowindex="0"
                      data-value="3.125"
                      role="cell"
                      style="min-width: 100px; max-width: 100px; line-height: 51px; min-height: 52px; max-height: 52px;"
                      tabindex="-1"
                    >
                      %3.125
                    </div>
                    <div
                      class="MuiDataGrid-cell MuiDataGrid-cellLeft"
                      role="cell"
                      style="min-width: 500px; max-width: 500px; line-height: 51px; min-height: 52px; max-height: 52px;"
                    />
                  </div>
                  <div
                    aria-rowindex="3"
                    aria-selected="false"
                    class="MuiDataGrid-row Mui-odd"
                    data-id="2"
                    data-rowindex="1"
                    role="row"
                    style="max-height: 52px; min-height: 52px;"
                  >
                    <div
                      aria-colindex="0"
                      class="MuiDataGrid-cell MuiDataGrid-cellLeft"
                      data-field="lenderName"
                      data-rowindex="1"
                      data-value="TFB Federal Credit Union"
                      role="cell"
                      style="min-width: 250px; max-width: 250px; line-height: 51px; min-height: 52px; max-height: 52px;"
                      tabindex="-1"
                    >
                      TFB Federal Credit Union
                    </div>
                    <div
                      aria-colindex="1"
                      class="MuiDataGrid-cell MuiDataGrid-cellLeft"
                      data-field="loanType"
                      data-rowindex="1"
                      data-value="10/1 ARM"
                      role="cell"
                      style="min-width: 250px; max-width: 250px; line-height: 51px; min-height: 52px; max-height: 52px;"
                      tabindex="-1"
                    >
                      10/1 ARM
                    </div>
                    <div
                      aria-colindex="2"
                      class="MuiDataGrid-cell MuiDataGrid-cellLeft"
                      data-field="interestRate"
                      data-rowindex="1"
                      data-value="3.25"
                      role="cell"
                      style="min-width: 100px; max-width: 100px; line-height: 51px; min-height: 52px; max-height: 52px;"
                      tabindex="-1"
                    >
                      %3.25
                    </div>
                    <div
                      class="MuiDataGrid-cell MuiDataGrid-cellLeft"
                      role="cell"
                      style="min-width: 500px; max-width: 500px; line-height: 51px; min-height: 52px; max-height: 52px;"
                    />
                  </div>
                  <div
                    aria-rowindex="4"
                    aria-selected="false"
                    class="MuiDataGrid-row Mui-even"
                    data-id="3"
                    data-rowindex="2"
                    role="row"
                    style="max-height: 52px; min-height: 52px;"
                  >
                    <div
                      aria-colindex="0"
                      class="MuiDataGrid-cell MuiDataGrid-cellLeft"
                      data-field="lenderName"
                      data-rowindex="2"
                      data-value="TFB Federal Credit Union"
                      role="cell"
                      style="min-width: 250px; max-width: 250px; line-height: 51px; min-height: 52px; max-height: 52px;"
                      tabindex="-1"
                    >
                      TFB Federal Credit Union
                    </div>
                    <div
                      aria-colindex="1"
                      class="MuiDataGrid-cell MuiDataGrid-cellLeft"
                      data-field="loanType"
                      data-rowindex="2"
                      data-value="30YR Fixed"
                      role="cell"
                      style="min-width: 250px; max-width: 250px; line-height: 51px; min-height: 52px; max-height: 52px;"
                      tabindex="-1"
                    >
                      30YR Fixed
                    </div>
                    <div
                      aria-colindex="2"
                      class="MuiDataGrid-cell MuiDataGrid-cellLeft"
                      data-field="interestRate"
                      data-rowindex="2"
                      data-value="3.5"
                      role="cell"
                      style="min-width: 100px; max-width: 100px; line-height: 51px; min-height: 52px; max-height: 52px;"
                      tabindex="-1"
                    >
                      %3.5
                    </div>
                    <div
                      class="MuiDataGrid-cell MuiDataGrid-cellLeft"
                      role="cell"
                      style="min-width: 500px; max-width: 500px; line-height: 51px; min-height: 52px; max-height: 52px;"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Mui-resizeTriggers"
    >
      <div
        class="expand-trigger"
      >
        <div
          style="width: 1px; height: 1px;"
        />
      </div>
      <div
        class="contract-trigger"
      />
    </div>
  </div>,
  " ",
]
`;
